server/src/main/java/org/elasticsearch/indices/recovery/RecoverySettings.java

```diff
@@ -46,49 +46,96 @@
 
     private static final Logger logger = LogManager.getLogger(RecoverySettings.class);
 
-    public static final Setting<ByteSizeValue> INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING = Setting.byteSizeSetting(
-        "indices.recovery.max_bytes_per_sec",
-        s -> {
-            final ByteSizeValue defaultMaxBytesPerSec = new ByteSizeValue(40, ByteSizeUnit.MB);
-            final List<DiscoveryNodeRole> roles = NodeRoleSettings.NODE_ROLES_SETTING.get(s);
-            final List<DiscoveryNodeRole> dataRoles = roles.stream()
-                .filter(DiscoveryNodeRole::canContainData)
-                .collect(Collectors.toUnmodifiableList());
-            if (dataRoles.isEmpty()) {
+    /**
+     * Disk's write bandwidth allocated for this node. When both this setting and {@link #NODE_NETWORK_AVAILABLE_BANDWIDTH_SETTING}
+     * are defined they are used to adjust the default value for {@link #INDICES_RECOVERY_MAX_BYTES_PER_SEC_SETTING} per node.
+     */
+    public static final Setting<ByteSizeValue> NODE_DISK_AVAILABLE_BANDWIDTH_SETTING = Setting.byteSizeSetting(
```

Comment:
I wonder if we want both read and write bandwidth here?

It does complicate things a bit, since we may then need more factors (a generic, a read and a write factor), but if we have those available, we can better change our defaults moving forward.

Also, I think we could clarify that this number should be for 1/2MB reads or writes, matching our default chunk size. I think we may want to add more of those in the future, but I think it is fine to stick to one for now.

Reply:
I added read & write bandwidths along with their corresponding factors. Let me know if that's ok.